src/uu/free/src/free.rs

```diff
@@ -101,88 +135,157 @@ fn parse_meminfo() -> Result<MemInfo, Box<dyn std::error::Error>> {
+where
+    F: Fn(u64) -> String,
+{
+    // ugly hack to convert negative values
+    let free_str: String = if free < 0 {
+        "-".to_owned() + &f((-free) as u64)
+    } else {
+        f(free as u64)
+    };
+
+    println!("{:8}{:>12}{:>12}{:>12}", name, f(total), f(used), free_str);
+}
+
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     let matches = uu_app().try_get_matches_from(args)?;
-    let wide = matches.get_flag("wide");
 
+    let wide = matches.get_flag("wide");
     let human = matches.get_flag("human");
-
     let si = matches.get_flag("si");
+    let total = matches.get_flag("total");
+    let lohi = matches.get_flag("lohi");
+    let count_flag = matches.get_one("count");
+    let mut count: u64 = count_flag.unwrap_or(&1_u64).to_owned();
+    let seconds_flag = matches.get_one("seconds");
+    let seconds: f64 = seconds_flag.unwrap_or(&1.0_f64).to_owned();
+    let committed = matches.get_flag("committed");
+    let one_line = matches.get_flag("line");
 
+    let dur = Duration::from_nanos(seconds.mul(1_000_000_000.0).round() as u64);
     let convert = detect_unit(&matches);
 
-    match parse_meminfo() {
-        Ok(mem_info) => {
-            let buff_cache = if wide {
-                mem_info.buffers
-            } else {
-                mem_info.buffers + mem_info.cached
-            };
-            let cache = if wide { mem_info.cached } else { 0 };
-            let used = mem_info.total - mem_info.available;
-
-            if wide {
-                wide_header();
-                if human {
-                    println!(
-                        "{:8}{:>12}{:>12}{:>12}{:>12}{:>12}{:>12}{:>12}",
-                        "Mem:",
-                        humanized(mem_info.total, si),
-                        humanized(used, si),
-                        humanized(mem_info.free, si),
-                        humanized(mem_info.shared, si),
-                        humanized(buff_cache, si),
-                        humanized(cache + mem_info.reclaimable, si),
-                        humanized(mem_info.available, si),
-                    )
+    let infinite: bool = count_flag.is_none() && seconds_flag.is_some();
+
+    while count > 0 || infinite {
+        // prevent underflow
+        if !infinite {
+            count -= 1;
+        }
+        match parse_meminfo() {
+            Ok(mem_info) => {
+                let buff_cache = if wide {
+                    mem_info.buffers
                 } else {
+                    mem_info.buffers + mem_info.cached
+                };
+                let cache = if wide { mem_info.cached } else { 0 };
+                let used = mem_info.total - mem_info.available;
+
+                // function that converts the number to the correct string
+                let n2s = |x| match human {
+                    true => humanized(x, si),
+                    false => convert(x).to_string(),
+                };
```

Comment:
I suggest to use an `if/else` instead of a `match` as it is more idiomatic.

Otherwise, I think the PR is ready to merge.

Reply:
fixed